### PR TITLE
glib: add components

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **glib/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
generated file has `Version: None`. this line is generated by https://github.com/conan-io/conan/blob/develop/conans/client/generators/pkg_config.py#L111
DYT it should use `self.conanfile.version` instead of `cpp_info.version` ? what is this cpp_info.version attribute ?

File: recipes/glib/all/conanfile.py [hunk 1]
```diff
@@ -136,17 +136,52 @@ def package(self):
...
+        if self.settings.os == "Linux":
+            if self.options.with_mount:
+                self.cpp_info.components["gio-2.0"].requires.append("libmount::libmount")
+            if self.options.with_selinux:
+                self.cpp_info.components["gio-2.0"].requires.append("libselinux::libselinux")
+
+        self.cpp_info.components["gresource"].libs = [] # this is actualy an executable
+        self.cpp_info.components["gresource"].requires.append("libelf::libelf") # this is actualy an executable
+
+        self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
```
Comment (uilianries) [inline recipes/glib/all/conanfile.py:187]:
```suggestion
        bin_path = os.path.join(self.package_folder, "bin")
        self.output.info("Appending PATH env var with: {}".format(bin_path))
        self.env_info.PATH.append(bin_path)
```

File: recipes/glib/all/test_package/conanfile.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "cmake", "pkg_config"
```
Comment (uilianries) [inline recipes/glib/all/test_package/conanfile.py:7]: You could exercise the CMake file with glib new targets from components

**Comment by ericLemanissier** [comment]
this one will pass when conan will be upgrade to 1.28.1

**Comment by uilianries** [comment]
> this one will pass when conan will be upgrade to 1.28.1

it should be done in 1 or 2 weeks ...

**Comment by ericLemanissier** [comment]
I reverted the components pkg-config test, so that it passes on conan 1.27, and can be merged without waiting conan 1.28.1 to be deployed on CCI

**Pull request merged**